Add Altium sponsor

cameronkinsella opened the pull request:
<!--
Hey! Thanks for creating a PR.
Please fill out the template below to make our review easier.
-->

## Description

Added Altium to sponsor list and updated the competition name.

## Type of change

<!--Add X in the appropriate box-->

- [X] Bug or typo fix
- [X] New feature

cameronkinsella committed:
Add Altium sponsor

cameronkinsella committed:
Update competition name

cameronkinsella committed:
List sponsors in alphabetical order

ABarroso647 commented:
I think there may be value in alphabetizing the names of the sponsors for readability

cameronkinsella committed:
Change Faculty of Engineering sponsor name

Changed the name to be in line with the other Waterloo sponsors.
Since the sponsor list is now sorted alphabetically, this will
ensure that all the Waterloo sponsors are grouped together.

ABarroso647 reviewed (approved):
LGTM :shipit:
